src/mqt/bench/cli.py

```diff
@@ -23,56 +26,127 @@ def format_help(self) -> str:
         return help_message + version_info
 
 
-def main() -> None:
-    """Invoke :func:`get_benchmark` exactly once with the specified arguments.
+def parse_benchmark_name_and_instance(algorithm: str) -> tuple[str, str | None]:
+    """Parse an algorithm name like "shor_xlarge" into a benchmark and instance name."""
+    if algorithm.startswith("shor_"):
+        parts = algorithm.split("_", 1)
+        return parts[0], parts[1]
+    return algorithm, None
+
 
-    The resulting QASM string is printed to the standard output stream;
-    no other output is generated.
-    """
+def main() -> None:
+    """Generate a single benchmark and output in specified format."""
     parser = CustomArgumentParser(description="Generate a single benchmark")
     parser.add_argument(
         "--level",
         type=str,
-        help='Level to generate benchmarks for ("alg", "indep", "nativegates" or "mapped")',
+        choices=["alg", "indep", "nativegates", "mapped"],
+        help='Level to generate benchmarks for ("alg", "indep", "nativegates" or "mapped").',
+        required=True,
+    )
+    parser.add_argument(
+        "--algorithm",
+        type=str,
+        help="Name of the benchmark (e.g., 'grover-v-chain', 'shor_xsmall').",
         required=True,
     )
-    parser.add_argument("--algorithm", type=str, help="Name of the benchmark", required=True)
-    parser.add_argument("--num-qubits", type=int, help="Number of Qubits", required=True)
-    parser.add_argument("--qiskit-optimization-level", type=int, help="Qiskit compiler optimization level")
-    parser.add_argument("--native-gate-set", type=str, help="Name of the provider")
-    parser.add_argument("--device", type=str, help="Name of the device")
+    parser.add_argument(
+        "--num-qubits",
+        type=int,
+        help="Number of qubits for the benchmark.",
+        required=True,
+    )
+    parser.add_argument(
+        "--qiskit-optimization-level",
+        type=int,
+        help="Qiskit compiler optimization level (0-3).",
+    )
+    parser.add_argument(
+        "--native-gate-set",
+        type=str,
+        help="Provider name for native gate set (e.g., 'ibm', 'rigetti').",
+    )
+    parser.add_argument(
+        "--device",
+        type=str,
+        help="Device name for mapping stage (e.g., 'ibm_washington').",
+    )
+    parser.add_argument(
+        "--output-format",
+        type=str,
+        choices=[fmt.value for fmt in OutputFormat],
+        default=OutputFormat.QASM2.value,
```

Comment:
```suggestion
        default=OutputFormat.QASM3.value,
```
probably more future proof